portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -48,35 +47,56 @@ private String getParameter(HttpServletRequest request, String name, String defa
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException{
 
-    String commentStr = getParameter(request, "comments-input", "");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     
-    if (commentStr.equals(""))
-      return;
+    int prevMaxCmt = maxComments;
+    maxComments = Integer.parseInt(getParameter(request, "cmt-max-input", "-1"));
+
+    if (maxComments == -1){
+      maxComments = prevMaxCmt;
+    }
+
+    String commentStr = getParameter(request, "comments-input", "");
     
-    Entity commentEntity = new Entity("Comment");
-    commentEntity.setProperty("comment_msg", commentStr);
+    if (!commentStr.equals("")){
+      Entity commentEntity = new Entity("Comment");
+      commentEntity.setProperty("comment_msg", commentStr);
 
-    datastore.put(commentEntity);
+      datastore.put(commentEntity);
+    }
 
-    response.sendRedirect("/index.html");
+    response.sendRedirect("/");
   }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    
     List<String> commentsList = new ArrayList();
 
     Query commentsQuery = new Query("Comment");
     PreparedQuery results = datastore.prepare(commentsQuery);
 
     for (Entity entity : results.asIterable()){
-      commentsList.add((String) entity.getProperty("comment"));
+      commentsList.add((String)entity.getProperty("comment_msg"));
+    }
+
+    // Shirk ArrayList based on max comments
+    if (maxComments >= 0 && maxComments < commentsList.size()){
```

Comment:
I think this might be a bit easier to follow with just a single `if / else if`: you could check first if `maxComments` is 0 and then if it's greater than 0 but less than `commentsList.size()`.

Reply:
Done. 👍